generators/add-events/cloud-events/index.js

```diff
@@ -0,0 +1,98 @@
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const path = require('path')
+const ActionGenerator = require('../../../lib/ActionGenerator')
+const { commonDependencyVersions } = require('../../../lib/constants')
+
+class CloudEventsGenerator extends ActionGenerator {
+  constructor (args, opts) {
+    super(args, opts)
+    this.props = {
+      description: 'This is a sample action showcasing how to create a cloud event and publish to I/O Events',
+      // eslint-disable-next-line quotes
+      requiredParams: `['organizationId', 'apiKey', 'providerId', 'eventCode', 'payload']`,
+      // eslint-disable-next-line quotes
+      importCode: `const { Events } = require('@adobe/aio-sdk')
+const cloudEventV1 = require('cloudevents-sdk/v1')`,
+      utilFunction: `
+function createCloudEvent(providerId, eventCode, payload) {
+  let cloudevent = cloudEventV1.event()
+    .data(payload)
+    .source('urn:uuid:' + providerId)
+    .type(eventCode)
+    .id('randomId')
+  return cloudevent.format()
+}
+      `,
+      responseCode: `
+    // initialize the client
+    const eventsClient = await Events.init(params.organizationId, params.apiKey, token)
+
+    // Create cloud event for the given payload
+    const cloudEvent = createCloudEvent(params.providerId, params.eventCode, params.payload)
+
+    // Publish to I/O Events
+    const published = await eventsClient.publishEvent(cloudEvent)
+    let statusCode = 200
+    if (published === 'OK') {
+      logger.info('Published successfully to I/O Events')
+    } else if (published === undefined) {
+      logger.info('Published to I/O Events but there were not interested registrations')
+      statusCode = 204
+    } 
+    const response = {
+      statusCode: statusCode,
+    }`
+    }
+  }
+
+  async prompting () {
+    this.props.actionName = await this.promptForActionName(
+      'creates messages in Cloud events format and publishes to Adobe I/O Events',
+      'events')
+  }
+
+  writing () {
+    // this.registerTransformStream(beautify({ indent_size: 2 }))
+    this.sourceRoot(path.join(__dirname, '../../templates'))
+
+    this.addAction(this.props.actionName, './stub-action.js', {
+      testFile: '../add-events/cloud-events/templates/publishEvents.test.js',
+      sharedLibFile: './utils.js',
+      sharedLibTestFile: './utils.test.js',
+      e2eTestFile: './stub-action.e2e.js',
+      tplContext: this.props,
+      dependencies: {
+        '@adobe/aio-sdk': commonDependencyVersions['@adobe/aio-sdk'],
+        'cloudevents-sdk': '^1.0.0',
+        uuid: '^8.0.0'
+      },
+      dotenvStub: {
+        label: 'please provide your Adobe I/O Events organization id and api key',
+        vars: [
+          'EVENTS_ORGANIZATION_ID',
+          'EVENTS_API_KEY'
```

Comment:
@sangeetha5491 there is one more change to do 😅  We've just merged a PR (https://github.com/adobe/generator-aio-app/pull/57) where all actions template use now a unique SERVICE_API_KEY env variable that is globally set. So you can remove 'EVENTS_API+KEY' from the list and in the `actionManifestConfig` you should change `apiKey: '$EVENTS_API_KEY'` for `apiKey: '$SERVICE_API_KEY'` 🙏

Reply:
done